source/common/router/scoped_rds.cc

```diff
@@ -227,40 +228,40 @@ void ScopedRdsConfigSubscription::onConfigUpdate(
   // NOTE: deletes are done before adds/updates.
 
   absl::flat_hash_map<std::string, ScopedRouteInfoConstSharedPtr> to_be_removed_scopes;
+  // Destruction of resume_rds will lift the floodgate for new RDS subscriptions.
+  // Note in the case of partial acceptance, accepted RDS subscriptions should be started
+  // despite of any error.
+  ScopedResume resume_rds;
   // If new route config sources come after the local init manager's initialize() been
   // called, the init manager can't accept new targets. Instead we use a local override which will
   // start new subscriptions but not wait on them to be ready.
   std::unique_ptr<Init::ManagerImpl> noop_init_manager;
-  // NOTE: This should be defined after noop_init_manager as it depends on the
-  // noop_init_manager.
-  std::unique_ptr<Cleanup> resume_rds;
+  // NOTE: This should be defined after noop_init_manager and resume_rds, as it depends on the
+  // noop_init_manager, and we want a single RDS discovery request to be sent to management server.
+  std::unique_ptr<Cleanup> init_noop_init_manager;
+  ASSERT(localInitManager().state() > Init::Manager::State::Uninitialized);
```

Comment:
What changed to make this an invariant vs. `if` statement?

Reply:
if we have gone this far, localinitMgr has its initialize() called already. 
The local_init_target will not be ready until the first onConfigUpdate return, so for the first batch of RDS subscriptions they can be part of the initialization of the envoy, and after that it will be the disposable_manager_ taking care of the rds initialization.